### DISCUSSION TITLE
TF-IDF and network construction

### DISCUSSION THREAD

**Discussion by BenxiaHu** [sankaranlab/SCAVENGE#5] [Q&A] [answered]
Hello Fulong,
    SCAVENGE is a good tool deciphering the function of genetic variant at single-cell level.
    I have 2 questions about the algorithm of SCAVENGE.
    1: what is the binarized sparse matrix?
    2: you used TF-IDF to calculate the weight for each feature. it seems that the IDF in your paper looks a little different from (log(N/(dfi+1))). 

<img width="543" alt="image" src="https://user-images.githubusercontent.com/12387462/173659148-dbae85fa-19a3-4d1c-b0b9-ccc8fc7b7866.png">

**Comment by fl-yu** [comment]
Hi, Thank you for your interest in our tool!

1. binarized sparse matrix is the read count matrix of feature(peak)-by-cell with all the counts more than 1 being valued as 1, as such, this matrix is binarized and the values in the matrix are either 1 or 0.

2. Yes, it is slightly different than standard idf. This version of idf calculation is commonly used in dimension reduction of scATAC-seq and performs well, for example, [Cusanovich2018](https://www.sciencedirect.com/science/article/pii/S0092867418308559) and [Stuart2021](https://www.nature.com/articles/s41592-021-01282-5). Actually, there are many variants in tf-idf calculation [link](https://plumbr.io/blog/programming/applying-tf-idf-algorithm-in-practice), I think the performance will be slightly different although I do not have a comprehensive benchmark.

Hope this is helpful. Please let me know if you have any other questions, Thanks!

**Comment by BenxiaHu** [comment]
thanks a lot for your explanation.
would you like to explain a little of how you build a nearest neighbor graph from the LSI matrix of N cells and d leading LSIs (d = 30)?
in your paper, it seems that you did not mention how to obtain LSI matrix. maybe I miss some important steps.
Best,

**Comment by fl-yu** [comment]
Great Happy to explain! I will convert it to the discussion as this is not an issue.

**Comment by fl-yu** [comment, marked as answer]
LSI matrix is calculated from the binarized count matrix by using TF-IDF and SVD techniques. Next, for each cell, the neighbor is evaluated by euclidean metric from LSI matrix and the top k cells were selected as the k-nearest neighbors (knn). We used a mutual knn graph rather than the original one by considering only the cells are the top k-nearest neighbors to each other, where we reasoned that it is more rigorous for cell-to-cell network construction.

**Comment by BenxiaHu** [comment]
hello Fulong,
     I really appreciate your explanation.
     I almost understand how to build the M-kNN.
     Another question: do you know the principle of gchromVAR? 
     I try to read the method the authors described.  (https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-019-0362-6/MediaObjects/41588_2019_362_MOESM1_ESM.pdf)
     
<img width="644" alt="image" src="https://user-images.githubusercontent.com/12387462/174099899-47237e2f-e54d-42ef-82ef-d7d3cd12979b.png">

<img width="401" alt="image" src="https://user-images.githubusercontent.com/12387462/174100490-6350508d-e5b1-4574-8ba2-3a137dec06ae.png">

it is not clear how do generate E matrix and correct confounding factors.

Best,
